File: src/main/java/com/wesker926/leetcode/algorithms/p1816/s1/solution.go
// Package s1
// @author: wesker.gh
// @date: 2021/12/6
// @description: 遍历
package s1

func truncateSentence(s string, k int) string {
	result := make([]byte, 0, len(s))
	for _, c := range s {
		if c == ' ' {
			k--
			if k == 0 {
				break
			}
		}
		result = append(result, byte(c))
	}
	return string(result)
}

File: src/main/java/com/wesker926/leetcode/algorithms/p1816/s1/Solution.java
package com.wesker926.leetcode.algorithms.p1816.s1;

/**
 * @author wesker.gh
 * @date 2021/12/6
 * @description 遍历
 */
public class Solution {
    public String truncateSentence(String s, int k) {
        StringBuilder sb = new StringBuilder();
        for (char c : s.toCharArray()) {
            if (c == ' ') {
                k--;
                if (k == 0) {
                    break;
                }
            }
            sb.append(c);
        }
        return sb.toString();
    }
}
